src/buildx/bake.ts

```diff
@@ -178,27 +178,197 @@ export class Bake {
   }
 
   public static parseDefinition(dt: string): BakeDefinition {
-    return <BakeDefinition>JSON.parse(dt);
+    const definition = <BakeDefinition>JSON.parse(dt);
+
+    // convert to composable attributes: https://github.com/docker/buildx/pull/2758
+    for (const name in definition.target) {
+      const target = definition.target[name];
+      if (target['cache-from'] && Array.isArray(target['cache-from'])) {
```

Comment:
I think this means that you can mix the csv style with the composable style but I just wanted to make sure I was reading this right.

Reply:
It means we always want composable attributes so if csv values are passed we convert them to composable attributes.